File: fitrbackend/src/main/java/com/example/fitrbackend/dto/CreateUserProfileRequest.java
package com.example.fitrbackend.dto;

import com.example.fitrbackend.model.ExperienceLevel;
import com.example.fitrbackend.model.Gender;
import com.example.fitrbackend.model.Goal;
import com.example.fitrbackend.model.Unit;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class CreateUserProfileRequest {
    private Gender gender;
    private float height;
    private float weight;
    private ExperienceLevel experienceLevel;
    private Goal goal;
    private Unit preferredWeightUnit;
    private Unit preferredDistanceUnit;
}

File: fitrfrontend/fitrfrontend/Models/DTO/CreateUserProfileRequest.swift
//
//  CreateUserProfileRequest.swift
//  fitrfrontend
//
//  Created by Ambrose Blay on 2/17/26.
//

import Foundation

struct CreateUserProfileRequest: Codable {
    let gender: Gender
    let height: Float
    let weight: Float
    let experienceLevel: ExperienceLevel
    let goal: Goal
    let preferredWeightUnit: Unit
    let preferredDistanceUnit: Unit
}
